### PR TITLE
Christa's latest system; Either; Maybe; example of Maybe from Wouter paper

### PR DESCRIPTION
**Pull request by haroldcarr**

### CODE REVIEW

**Review by mark-moir — COMMENTED** [review]
I've made a number of comments from a quick review.  However, my overall impression is that this is not ready to be merged into main.  I suggest that we establish a new branch (e.g., `main-ast`), where we continue to get this work ready for merging.

File: src/Dijkstra/AST/Example.agda [hunk 1]
```diff
@@ -0,0 +1,269 @@
+{- Byzantine Fault Tolerant Consensus Verification in Agda, version 0.9.
+
+   Copyright (c) 2022, Oracle and/or its affiliates.
+   Licensed under the Universal Permissive License v 1.0 as shown at https://opensource.oracle.com/licenses/upl
+-}
+
+{-# OPTIONS --allow-unsolved-metas #-}
+
+module Dijkstra.AST.Example (Ev Wr St : Set) where
```
[the remark this answers is not in the document]
Comment (haroldcarr) [reply src/Dijkstra/AST/Example.agda:9]: That is a question for @cwjnkins

File: src/Dijkstra/AST/Example.agda [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            (ASTOpSem.runAST RWSOpSem
+             ((λ { (Level.lift nothing) → ASTreturn (unit , (λ x₁ → x₁ ++ x₁))
+                 ; (Level.lift (just x))
+                     → ASTbind (ASTop (RWStell (x ∷ []) refl) (λ ()))
+                       (λ _ → ASTreturn (unit , (λ _ → [])))
+                 })
+              (Level.lift (f s)))
+             (e , s)))))
+         ≡ 0)
+-}
```
[the remark this answers is not in the document]
Comment (haroldcarr) [reply src/Dijkstra/AST/Example.agda:208]: @cwjnkins

File: src/Dijkstra/AST/Example.agda [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+-- 34 lines...
+    --   with f s
+    -- ... | nothing = refl , refl
+    -- ... | just x  = refl , refl
+
+
+  -- wpTwoOuts : ∀ f i → ASTPredTrans.predTrans RWSPT (prog₁ f) TwoOuts i
+  -- wpTwoOuts f (e , s) w w= unit refl .(w ∷ w ∷ []) refl = refl
+
+-- RWSPTMono : ASTPredTransMono RWSPT
```
[the remark this answers is not in the document]
Comment (haroldcarr) [reply src/Dijkstra/AST/Example.agda:218]: @cwjnkins

File: src/Dijkstra/AST/Example.agda (hunk 2 above)
[the remark this answers is not in the document]
Comment (cwjnkins) [reply src/Dijkstra/AST/Example.agda:210]: @mark-moir You can remove `Example.agda` from being tracked, I am only using it for code listings in the paper.

File: src/Dijkstra/AST/Example.agda (hunk 3 above)
[the remark this answers is not in the document]
Comment (cwjnkins) [reply src/Dijkstra/AST/Example.agda:220]: @mark-moir You can remove Example.agda from being tracked, I am only using it for code listings in the paper.

[the remark this answers is not in the document]
Comment (haroldcarr) [reply src/Dijkstra/AST/Example.agda]: `git rm Example.agda`
DONE

File: src/Dijkstra/AST/Maybe.agda [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+--  ASTSufficientPT.sufficient MaybeSuf (prog₁ a) BailWorks unit (bailWorks a unit)
+  bailWorksSuf a i
+    with runMaybe (prog₁ a) i
+  ... | x≡x = refl
+
+  postulate
+    maybeNeccessary
+      : ∀ {A} (m : MaybeD A) P i → P (runMaybe m i) → predTrans m P i
+
+    maybePTApp
```
[the remark this answers is not in the document]
Comment (haroldcarr) [reply src/Dijkstra/AST/Maybe.agda:166]: @cwjnkins

**Comment by mark-moir** [comment]
Thanks @haroldcarr.  I suggest that we merge #203 and then #205 and then finally make a `main-ast` branch out of the result.  This one is APPROVED.  Thanks.

**Pull request closed without merging**